analysis/11_an_stand_surv_asthma.do

```diff
@@ -0,0 +1,78 @@
+/*==============================================================================
+DO FILE NAME:			11_an_stand_surv_asthma
+PROJECT:				ICS in COVID-19 
+AUTHORS:				A Schultze
+						adapted from K Baskharan A Wong
+DATE: 					12th of May 2020
+VERSION: 				Stata 16.1 
+DESCRIPTION OF FILE:	fit parametric model
+						use postestimation command to plot standardised surv 
+						estimate difference in % at 2 months
+						derive marginal HR
+						
+DATASETS USED:			$tempdir\analysis_dataset_STSET_$outcome.dta
+DATASETS CREATED: 		None
+OTHER OUTPUT: 			Results in svg: $outdir\standsurvplot1
+						Log file: $logdir\11_an_stand_surv_asthma 
+						
+==============================================================================*/
+
+* Open a log file
+capture log close
+log using $logdir\11_an_stand_surv_asthma, replace t
+
+* Open Stata dataset
+use $tempdir\analysis_dataset_STSET_$outcome, clear
+
+/* Sense check outcomes=======================================================*/ 
+tab exposure $outcome
+
+/* Fit the parametric model===================================================*/
+
+xi i.exposure i.male $varlist
+stpm2 _I* age1 age2 age3, scale(hazard) df(3) eform nolog stratify(stp)
```

Comment:
This is from @krishnanbhaskaran when I did the NSAID study: 
`stratify, which sort of does similar to stcox stratify, in that it allows the baseline hazard to differ by the stratification variable, but in stpm2, the baseline hazard is actually estimated, so this ends up meaning that you are asking Stata to fit an interaction between the underlying timescale (spline) and stp - it may be difficult to fit. If so, I think prob ok to drop the stratification just for the purposes of this plot.`

Reply:
This makes sense; I've dropped as well as it made little difference in my main analyses.